### DISCUSSION TITLE
Column-level authority control

### DISCUSSION THREAD

**Discussion by yongjinhou** [ClickHouse/ClickHouse#52342] [Q&A] [answered]
Does ClickHouse support column-level authority control? If yes, could you provide corresponding pr or design document for learning? I don't seem to have found it. And ClickHouse appears to provide only a column-level authentication interface.

**Comment by UnamedRus** [comment]
Can you explain what do you mean under column-level authority control?

May be link to some other dbms documentation about that feature

**Reply by yongjinhou** [reply]
The permission levels of database generally include GLOBAL (authorization to the whole world), DATABASE (authorization to database) and TABLE (authorization to table). I want to know whether ClikHouse supports COLUMN level permission level and can authorize columns, and whether there is any relevant pr or documentation.

**Reply by UnamedRus** [reply, marked as answer]
Yeah, you can

https://clickhouse.com/docs/en/sql-reference/statements/grant#select

https://clickhouse.com/docs/en/sql-reference/statements/grant#granting-privilege-syntax

**Reply by yongjinhou** [reply]
thank you very much.
